packages/odyssey-react-mui/src/labs/TopNav.tsx

```diff
@@ -0,0 +1,430 @@
+/*!
+ * Copyright (c) 2022-present, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import styled from "@emotion/styled";
+import { memo, useMemo, ReactElement, useCallback, KeyboardEvent } from "react";
+
+import { Box } from "../Box";
+import type { HtmlProps } from "../HtmlProps";
+import {
+  AuraIcon,
+  AuraWordmarkIcon,
+  QuestionCircleIcon,
+  SettingsIcon,
+} from "../icons.generated";
+import { Link } from "../Link";
+import {
+  DesignTokens,
+  useOdysseyDesignTokens,
+} from "../OdysseyDesignTokensContext";
+
+export type TopNavLinkItem = {
+  id: string;
+  label: string;
+  /**
+   * link added to the nav item. if it is undefined, static text will be displayed.
+   * fires onClick event when it is passed
+   */
+  href?: string;
+  /**
+   * determines whether the link item is diabled
+   */
+  isDisabled?: boolean;
+  /**
+   * Event fired when the nav item is clicked
+   */
+  onClick?(): void;
+  /**
+   * The link target prop. e.g., "_blank"
+   */
+  target?: string;
+};
+
+export type UserProfileProps = {
+  /**
+   * Logged in user profile icon to be displayed in the top nav
+   */
+  profileIcon?: ReactElement;
+  /**
+   * Logged in user info to be displayed in the top nav
+   */
+  userName: string;
+  /**
+   * Org name of the logged in user
+   */
+  orgName: string;
+};
+
+export type TopNavProps = {
+  /**
+   * Determines whether to display the logo (aura & wordmark)
+   */
+  hasLogo?: boolean;
+  /**
+   *  Determines whether the search box is displayed
+   */
+  search?: ReactElement;
```

Comment:
I am guessing we can leave this prop as something the consuming application can pass through with their customizations. In this case, it'll be SearchField that can used as Global Search (in Admin) or App Search (in enduser dashboard) etc.

Reply:
I don't mind leaving the consumer to provide this component. If we are going to do that, I'd recommend we rename this prop to something like `SearchFieldComponent` though. Feel like that makes it more clear what it's expecting and how it will be used

Reply:
I'd make the description of the prop a little more verbose here also. Something about how they should be passing in a `SearchField` component

Reply:
Or, we create a new component specifically for the top nav like `TopNavSearchField` that sets the variant for them